setup/src/Magento/Setup/Validator/IpValidator.php

```diff
@@ -66,14 +66,32 @@ public function validateIps(array $ips, $noneAllowed)
      */
     private function filterIps(array $ips)
     {
-        foreach ($ips as $ip) {
-            if (filter_var($ip, FILTER_VALIDATE_IP)) {
-                $this->validIps[] = $ip;
-            } elseif ($ip == 'none') {
-                $this->none[] = $ip;
-            } else {
-                $this->invalidIps[] = $ip;
+        foreach ($ips as $range) {
+            if ($range === 'none') {
+                $this->none[] = $range;
+                continue;
             }
+
+            $cidr = 32;
+            $ip = $range;
+            if (strpos($range, '/') !== false) {
+                list($ip, $cidr) = explode('/', $range);
```

Comment:
cidr? Is there some existing logic in ZF2 or Symfony which can be reused?

Reply:
See https://github.com/magento/magento2/pull/10570#issuecomment-323649544

Reply:
Maybe there is, but I havent found it. And it seems overkill to include another class as replacement for a list = explode.

Reply:
I mean this logic in general - splitting by slash etc.

Reply:
Still not sure what "cidr" means, https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing#CIDR_notation looks like after splitting by slash there should be `$networkMask` instead.

Reply:
Ah that's what you mean. Yes you are right. I'll rename it.

Reply:
I'll call it `$subnetMask` as that is the name I heard most for this.